sdk/include/ds/xoroshiro.h

```diff
@@ -0,0 +1,75 @@
+// Copyright Microsoft and CHERIoT Contributors.
+// SPDX-License-Identifier: MIT
+//
+// Imported from snmalloc 4f9d991449380ed7d881a25ba02cc5668c1ff394.
+
+#pragma once
+
+#include <cstdint>
+#include <cstdlib>
+
+namespace ds::xoroshiro
+{
+	namespace detail
+	{
+		template<typename STATE, typename RESULT, STATE A, STATE B, STATE C>
```

Comment:
clang-tidy should tell you not to use these names, please can you check why not and fix it?  UPPERCASE should be reserved for MACROs (which may have side effects).

Reply:
I think it doesn't object to UPPERCASE as a degenerate form of CamelCase?  Even if I set
```
    - key: readability-identifier-naming.TemplateParameterCase
      value: CamelCase
```
`clang-tidy` doesn't object to all caps.

Reply:
Hmm, it does for other things.  Weird.